src/gnatss/ops/qc.py

```diff
@@ -0,0 +1,223 @@
+from typing import List, Optional, Tuple
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+from .. import constants
+from ..configs.main import Configuration
+from ..utilities.geo import calc_enu_comp
+from ..utilities.time import AstroTime
+
+# The 8 Colorblind from Bang Wong’s
+# Nature Methods paper https://www.nature.com/articles/nmeth.1618.pdf
+CB_COLORS: List[str] = [
+    "#000000",
+    "#E69F00",
+    "#56B4E9",
+    "#009E73",
+    "#F0E442",
+    "#0072B2",
+    "#D55E00",
+    "#CC79A7",
+]
+
+
+def _sec_to_iso(sec: float, fmt: str = "unix_j2000") -> str:
+    """Convert seconds to ISO format via AstroTime
+
+    Parameters
+    ----------
+    sec : float
+        The seconds to convert
+    fmt : str, default "unix_j2000"
+        The astropy time format
+
+    Returns
+    -------
+    str
+       The ISO formatted time string
+       as 'YYYY-MM-DDThh:mm:ss.dddddd'
+    """
+    astro_time = AstroTime(sec, format=fmt)
+    return astro_time.strftime("%Y-%m-%dT%H:%M:%S.%f")
+
+
+def _compute_ticks_and_labels(
+    data: pd.DataFrame, n_ticks: int = 25, time_col: str = "time"
+) -> Tuple:
+    """Compute ticks and labels for x-axis
+
+    Parameters
+    ----------
+    data : pd.DataFrame
+        The data to compute ticks and labels for
+        assuming that the data has a time column
+        and that the time column is in seconds
+    n_ticks : int, default 25
+        The desired number of ticks on the x-axis
+    time_col : str, default "time"
+        The name of the time column
+
+    Returns
+    -------
+    ticks : np.ndarray
+        The ticks values
+    labels : np.ndarray
+        The tick labels
+    """
+    time_min = data[time_col].min()
+    time_max = data[time_col].max()
+    step = (time_max - time_min) / n_ticks
+    ticks = np.arange(time_min, time_max + step, step)
+    labels = np.apply_along_axis(_sec_to_iso, 0, ticks)
+    return ticks, labels
+
+
+def plot_residuals(
+    residuals_df: pd.DataFrame,
+    outliers_df: Optional[pd.DataFrame] = None,
+    n_ticks: int = 15,
+    figsize: tuple = (10, 5),
+) -> plt.Figure:
+    """Plot residuals
+
+    Parameters
+    ----------
+    residuals_df : pd.DataFrame
+        Residuals dataframe
+    outliers_df : pd.DataFrame, optional
+        Outliers dataframe
+    n_ticks : int, default 15
+        The desired number of ticks on the x-axis
+    figsize : tuple, default (10, 5)
+        The figure size
+
+    Returns
+    -------
+    plt.Figure
+        The matplotlib figure
+    """
+    # Setup plot
+    fig, axs = plt.subplots(figsize=figsize)
+
+    # Plot residuals
+    time_labels = [constants.TIME_J2000, constants.TIME_ISO]
+    col_names = []
+    for i, col in enumerate(residuals_df.columns):
+        if col not in time_labels:
+            col_names.append(col)
+            residuals_df.plot.scatter(
+                x="time", y=col, s=2, ax=axs, c=CB_COLORS[i + 1], zorder=i + 2
```

Comment:
Any chance of going over the length of `CB_COLORS`?

Reply:
No, this should max out to only 6 total upstream.